waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -432,6 +394,44 @@ validateUserNodeVersionRange spec =
           ]
         else []
 
+validatePrismaSchema :: AppSpec -> [ValidationError]
+validatePrismaSchema spec =
+  concat
+    [ concatMap validateModel models,
+      validateGenerators generators,
+      validateDatasources datasources
+    ]
+  where
+    validateModel :: Psl.Ast.Model -> [ValidationError]
+    validateModel model@(Psl.Ast.Model modelName _) = concatMap validateModelField $ Psl.Ast.getModelFields model
+      where
+        validateModelField :: Psl.Ast.Field -> [ValidationError]
+        validateModelField (Psl.Ast.Field fieldName _type typeModifiers _attrs) = concatMap (validateTypeModifier fieldName) typeModifiers
+
+        validateTypeModifier :: String -> Psl.Ast.FieldTypeModifier -> [ValidationError]
```

Comment:
It's not really possible to throw error in the Parsec world and I wasn't able to build a system around "[]" not being followed by "?" since we wrap each parser with `try` and then if the parsing of model fields fails, the parsing of whole model fails and the parsing stops. Then we get an empty schema 😢 

It's better to parse the expected cases and then validate them here with proper error messages that user can act on IMHO.

Reply:
I think we will manage to parse it! In that case you will want to update this part, let's see.

Reply:
I've use `eof` parser to make the Prisma schema parser fail if the file wasn't fully parsed. This works well 👍 

It says
```
❌ --- [Error] Your wasp project failed to compile: -------------------------------

- Wasp couldn't parse your Prisma schema file, please check if you have any errors in it.

(line 17, column 21):
unexpected "?"
expecting identifier, "@" or "}"
```

And then in their IDE the Prisma LS kicks in:
![Screenshot 2024-06-18 at 23 39 36](https://github.com/wasp-lang/wasp/assets/2223680/4de2c37d-31f2-42a1-8fa5-21d2484722c7)